lib/Mojolicious/Guides/Testing.pod

```diff
@@ -0,0 +1,686 @@
+
+=encoding utf8
+
+=head1 NAME
+
+Mojolicious::Guides::Testing - Web Application Testing Made Easy
+
+=head1 OVERVIEW
+
+This document is an introduction to testing web applications with
+L<Test::Mojo>. L<Test::Mojo> can be thought of as a module that provides all
+of the tools and testing assertions needed to test web applications in a
+Perl-ish way.
+
+While L<Test::Mojo> can be used to test any web application, it has shortcuts
+designed to make testing L<Mojolicious> web applications easy and pain-free.
+
+Please refer to the L<Test::Mojo> documentation for a complete reference to
+many of the ideas and syntax introduced in this document.
+
+A test file for a simple web application might look like:
+
+  use Mojo::Base -strict;
+
+  use Test::Mojo;
+  use Test::More;
+
+  # Start a Mojolicious app named "Celestial"
+  my $t = Test::Mojo->new('Celestial');
+
+  # Post a JSON document
+  $t->post_ok('/notifications', json => {event => 'full moon', via => 'sms', channel => '+15555555555'})
+    ->status_is(201)
+    ->json_is('/message' => 'notification created');
+
+  # Perform GET requests and look at the responses
+  $t->get_ok('/sunrise')
+    ->status_is(200)
+    ->content_like(qr/ am$/);
+  $t->get_ok('/sunset')
+    ->status_is(200)
+    ->content_like(qr/ pm$/);
+
+  # Post a URL-encoded form
+  $t->post_ok('/insurance', form => {name => 'Jimmy', amount => '3.000.000', currency => '€'})
+    ->status_is(200);
+
+  # Use Test::More's like() to check the response
+  like $t->tx->res->dom->at('div#thanks')->text, qr/thank you/, 'thanks';
+
+  done_testing();
+
+In the rest of this document we'll explore these concepts and others related
+to L<Test::Mojo>.
+
+=head1 CONCEPTS
```

Comment:
Reading through the `head2` sections, it seems like they all fit within "concepts", but perhaps I can rethink how these are organized. Open to ideas.